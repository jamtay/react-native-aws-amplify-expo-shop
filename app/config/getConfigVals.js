import Constants from 'expo-constants';

export const getIsUsingMock = () => {
  return Constants.manifest.extra.isUsingMock || false;
};
