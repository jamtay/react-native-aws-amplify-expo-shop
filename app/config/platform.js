export const isAndroidOS = () => Platform.OS === 'android';
