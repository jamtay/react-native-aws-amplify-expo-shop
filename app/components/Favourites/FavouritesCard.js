import React from 'react';

import {View} from 'react-native'
import Result from '../SearchResults/Result';
import SearchResults from '../SearchResults';
import {isAndroidOS} from '../../config/platform';

/**
 * A wrapper component to display favourites with
 * @param width
 * @param item
 */
const FavouritesCard = ({width, item}) => {
  const isAndroid = isAndroidOS();
  return (
    <View testID={`favourite-${item.id}`}>
      <SearchResults>
        <Result
          width={width - 25}
          item={item}
          topStyle={isAndroid ? -32 : -20}
          isFavouriteCard
        />
      </SearchResults>
    </View>
  );
};

export default FavouritesCard;
