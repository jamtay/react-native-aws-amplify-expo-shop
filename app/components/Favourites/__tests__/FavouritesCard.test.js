import React from 'react';

import renderer from 'react-test-renderer';
import {render} from 'react-native-testing-library';
import {isAndroidOS} from '../../../config/platform';
import FavouritesCard from '../FavouritesCard';

jest.mock('../../../config/platform');
jest.mock('../../SearchResults', () => 'SearchResults');
jest.mock('../../SearchResults/Result', () => 'Result');

describe('FavouritesCard component tests', () => {
  const props = {
    width: 100,
    item: {
      id: 'id',
      name: 'name',
      description: 'description',
      fullAddress: 'fullAddress',
      addressLine1: 'addressLine1',
      addressLine2: 'addressLine2',
      addressLine3: 'addressLine3',
      postcode: 'postcode',
      country: 'country',
    },
  };

  beforeEach(() => {
    jest.resetAllMocks();
  });

  describe('snapshot test', () => {
    it('should match stored snapshot', () => {
      isAndroidOS.mockReturnValueOnce(true);
      const tree = renderer.create(<FavouritesCard {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('render with properties', () => {
    it('should render with correct width and item', async () => {
      isAndroidOS.mockReturnValueOnce(true);
      const {getByTestId} = render(<FavouritesCard {...props} />);
      const searchResultsComponent = await getByTestId('favourite-id').props
        .children;
      const resultProps = searchResultsComponent.props.children.props;

      expect(resultProps.item).toEqual(props.item);
      expect(resultProps.width).toEqual(props.width - 25);
    });

    it('should render with -32 topStyle for android', async () => {
      isAndroidOS.mockReturnValueOnce(true);
      const {getByTestId} = render(<FavouritesCard {...props} />);
      const searchResultsComponent = await getByTestId('favourite-id').props
        .children;
      const resultProps = searchResultsComponent.props.children.props;

      expect(resultProps.topStyle).toEqual(-32);
    });

    it('should render with -30 topStyle for ios', async () => {
      isAndroidOS.mockReturnValueOnce(false);
      const {getByTestId} = render(<FavouritesCard {...props} />);
      const searchResultsComponent = await getByTestId('favourite-id').props
        .children;
      const resultProps = searchResultsComponent.props.children.props;

      expect(resultProps.topStyle).toEqual(-20);
    });
  });
});
