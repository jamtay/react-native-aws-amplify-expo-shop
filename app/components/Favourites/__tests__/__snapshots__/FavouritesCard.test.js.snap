// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesCard component tests snapshot test should match stored snapshot 1`] = `
<View
  testID="favourite-id"
>
  <SearchResults>
    <Result
      isFavouriteCard={true}
      item={
        Object {
          "addressLine1": "addressLine1",
          "addressLine2": "addressLine2",
          "addressLine3": "addressLine3",
          "country": "country",
          "description": "description",
          "fullAddress": "fullAddress",
          "id": "id",
          "name": "name",
          "postcode": "postcode",
        }
      }
      topStyle={-32}
      width={75}
    />
  </SearchResults>
</View>
`;
